nixos/modules/programs/spacefm.nix

```diff
@@ -0,0 +1,41 @@
+# Global configuration for spacefm.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.programs.spacefm;
+
+in
+{
+  ###### interface
+
+  options = {
+
+    programs.spacefm = {
+
+      settings = mkOption {
+        type = types.attrs;
+        default = {};
+        example = literalExample ''{
```

Comment:
travis is probably not a good way to test modules, it only checks whether options evaluate. You should instead build a minimal NixOS system with the module enabled (if you're at the nixpkgs top-level and configuration.nix contains a minimal config, `nix-build nixos -I nixos-config=configuration.nix -A vm` should give you a runnable vm).